src/lib.rs

```diff
@@ -306,12 +341,13 @@ fn calculate_means<V: Value>(
 pub fn kmeans_lloyd<V: Value>(
     data: &ArrayView2<V>,
     k: usize,
+    n_trials: Option<usize>,
```

Comment:
~~This changes the behavior and structure of the algorithm sufficiently that I think it's better to have a separate implementation. Create a new `kmeans_lloyd_with_n_trials` function that takes the n_trials parameter.~~

As long as the original behavior is preserved when calling this function when `n_trials` is none I'm happy to drop this one. It might still be worth splitting the implementation of `initialize_plusplus` for the two different cases though.